rs/execution_environment/src/execution/update.rs

```diff
@@ -65,8 +65,9 @@ pub fn execute_update(
                     .as_ref()
                     .map_or(false, |es| es.is_wasm64);
 
-                let prepaid_execution_cycles =
-                    match round.cycles_account_manager.prepay_execution_cycles(
+                let prepaid_execution_cycles = match round
+                    .cycles_account_manager
+                    .prepay_execution_cycles(
```

Comment:
In the code below, we might also fail the execution when we create a new `UpdateHelper`:

```
let helper = match UpdateHelper::new(&clean_canister, &original, deallocation_sender) {
```

Seems we check there the memory limit: `wasm_memory_usage > wasm_memory_limit`. So if the memory usage is above the hook threshold and above the memory limit, the hook won’t be executed it seems. You can address this in a follow up PR...